### DISCUSSION TITLE
Is it safe to delete `fishd.tmp.******` files?

### DISCUSSION THREAD

**Discussion by TonalidadeHidrica** [fish-shell/fish-shell#9063] [Q&A]
There are many files named `fishd.tmp.******` under `~/.config/fish/`.  Most of them are empty, while a few of them seemingly contain universal variables in the same format as in `fish_variables`.  What are these for?  Are they safe to delete?

<img width="565" alt="image" src="https://user-images.githubusercontent.com/47710717/178188851-439a523c-40d5-43f1-bf4b-ff0e07661a91.png">

**Comment by krobelus** [comment]
On Sun, Jul 10, 2022 at 09:33:21PM -0700, TonalidadeHidrica wrote:
> There is a many files named `fishd.tmp.******` under `~/.config/fish/`.  [...] Are they safe to delete?

Yeah. fishd has been removed a long time ago (in version 2.1.0 from 2014)

**Reply by TonalidadeHidrica** [reply]
Hmm, that's strange, it's at least after 2018 that I started using this PC, I've always been using the latest available `fish` fetched from Homebrew, and these files seems to be updated within a year, some even within a month.  But well, I'll try removing them.  I'll ask you if new files are generated.  Thank you.

**Reply by faho** [reply]
Fishd has been gone since 2014, but the *file* was named fishd (with a mac address or hostname) until 3.0, where it was renamed "fish_variables" (without any machine identifier).

And 3.0.0 was released in december 2018.

----

HOWEVER: These are *tmp* files, "fishd.tmp.SOMERANDOMHASH". They are temporary names for when we create the file again.

Those are still named fishd, and we write them to a temporary directory, and try to clean them. It appears that hasn't always happened.

Tbh I recommend a temporary directory that is cleaned regularly, like on reboot.

**Reply by TonalidadeHidrica** [reply]
...but actually, the files are still being generated since 10 years ago, when I entirely deleted all these files, until now.  Is it possible that any extension generates such a file?

**Reply by faho** [reply]
@TonalidadeHidrica Please re-read my comment, I edited it 3 minutes after posting.

**Reply by TonalidadeHidrica** [reply]
So you mean that, still in fish 3, it temporality creates a file named `fishd.tmp.HASH` which, however, be indented to be renamed to `fish_variables`, right?  Then is this possibly an unintended behavior which should be reported as a bug?
Can it have something to do with my usage where `.config/fish/` is a symlink of a directory somewhere else?

**Reply by faho** [reply]
>So you mean that, still in fish 3, it temporality creates a file named fishd.tmp.HASH which, however, be indented to be renamed to fish_variables, right? Then is this possibly an unintended behavior which should be reported as a bug?

If you can still reproduce it, sure.

>Can it have something to do with my usage where .config/fish/ is a symlink of a directory somewhere else?

Probably not. We don't really do anything here that would be affected by it - we do "realpath" on the destination, but it should still remove the temp file even if it failed.

My best guess is this is aborted by a signal. Which means this is gonna be a pain to narrow down.

**Comment by ntwb** [comment]
Just wanting so somewhat +1 this issue, I'm not able to determine any reproducible steps to help debug this, but....

I reformatted my laptop and as part of this process I want to update my _dotfiles_ and repo.

So, as of ~2 weeks ago I reformatted my Macbook, the _only_ applications or software I have installed is via Home Brew

The rough order of processes since the reformat:

1. Reformat with macos
2. Install home brew
3. Install fish
4. Install other tools, apps, taps, casks etc

In those ~2 weeks since I've deleted ~20 `fishd.tmp.hash` files a few days ago, today I have 1 new file

If I can manage to come up with anything of substance to aid in reproducing this I shall be sure to update here

(I'll be reformatting my desktop Mac once I've the laptop setup final, so hopefully I can come up with some clean reproducible steps)

**Comment by ignamartinoli** [comment]
Hey, I'm facing the same issue. For some reason, my Fish config directory is plagued with these files.
Any news on this?
